Python/Movie-Info-Telegram-Bot/bot.py

```diff
@@ -0,0 +1,153 @@
+
+def error(update, context):
+    """Log Errors caused by Updates."""
+    logger.warning('Update "%s" caused error "%s"', update, context.error)
+
+def get_info(movie):
+    "To scrape IMDb and get genre and rating of the movie "
+    url = 'https://www.imdb.com/find?q='
+    r=requests.get(url+movie+'&ref_=nv_sr_sm')
+    soup=BeautifulSoup(r.text,"html.parser")
+    title = soup.find('title')
+    tags = soup('a')
+    pre_url=""
+    count=0
+    lis=[]
+    res=[]
+    for tag in tags:
+        if(count>2):
+            break
+        m=re.search('<a href=.*>(.*?)</a>',str(tag))
+        try:
+            lis=[]
+            link=re.search('/title/(.*?)/',str(m))
+            new_url='https://www.imdb.com'+str(link.group(0))
+            if new_url!=pre_url:
+                html=requests.get(new_url)
+                soup_each_title=BeautifulSoup(html.text,"html.parser")
+                movie_title=soup_each_title.find('title').string.replace('- IMDb',' ')
+                anchor=soup_each_title('a')
+                genre_string="Genre : "
+                for each in anchor:
+                    genre=re.search('<a href=\"/search/title\?genres=.*> (.*?)</a>',str(each))
+                    try:
+                       genre_string+=genre.group(1)+' '
+                    except:
+                        pass
+                strong_tag=soup_each_title('strong')
+                for i in strong_tag:
+                    rating=re.search('<strong title=\"(.*?) based',str(i))
+                    try:
+                        rating_string="IMDb Rating : "+rating.group(1)
+                    except:
+                        pass
+                details="For more details : "+new_url
+                lis.append(movie_title)
+                lis.append(genre_string)
+                lis.append(rating_string)
+                lis.append(details)
+                pre_url=new_url
+                count+=1
+                res.append(lis)
+        except :
+            pass
+    return res
+def main():
+    """Start the bot."""
+    # Create the Updater and pass it your bot's token.
+    # Make sure to set use_context=True to use the new context based callbacks
+    updater = Updater(TOKEN, use_context=True)
+
+    # Get the dispatcher to register handlers
+    dp = updater.dispatcher
+
+    # on different commands - reply in Telegram
+    dp.add_handler(CommandHandler("start", start))
+    dp.add_handler(CommandHandler("help", help))
+    dp.add_handler(CommandHandler("name", name))
+    dp.add_handler(CommandHandler("genre", genre))
+
+    # log all errors
+    dp.add_error_handler(error)
+
+    # Start the Bot
+    updater.start_polling()
+
+    # Run the bot until you press Ctrl-C or the process receives SIGINT,
+    # SIGTERM or SIGABRT. This should be used most of the time, since
+    # start_polling() is non-blocking and will stop the bot gracefully.
+    updater.idle()
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
Can we have an EOF here please

Reply:
Sure

Reply:
I tried using yapf and autopep8 and somehow got messed up. So I used a code formatter . @vybhav72954  kindly re-review the pr